go/enclave/rollupchain/rollup_chain.go

```diff
@@ -444,12 +440,16 @@ func (rc *RollupChain) SubmitBlock(block types.Block, isLatest bool) common.Bloc
 	}
 
 	common.TraceWithID(rc.nodeID, "Update state: b_%d", common.ShortHash(block.Hash()))
-	blockState, _, newLogs := rc.updateState(&block)
+	blockState := rc.updateState(&block)
 	if blockState == nil {
 		return rc.noBlockStateBlockSubmissionResponse(&block)
 	}
 
-	encryptedLogs, err := rc.subscriptionManager.GetSubscribedLogsEncrypted(newLogs, blockState.HeadRollup)
+	logs, found := rc.storage.FetchLogs(block.Hash())
+	if !found {
+		log.Panic("Could not retrieve logs for stored block state")
```

Comment:
Maybe just log an error and then proceed with empty logs? I guess it's a situation that should never happen but we don't want accidental bugs to bring down testnet.